### DISCUSSION TITLE
Issues with Image Blurriness and InscribedImage Constructor

### DISCUSSION THREAD

**Discussion by VADDIJASWANT** [clientIO/joint#2376] [Q&A] [answered]
### Introduction

I am writing to request your assistance with two issues I am encountering while using JointJS.

The first issue involves image blurriness when using the Image element. Specifically, when I create an element using var image = new joint.shapes.standard.Image(), any image that I add appears blurry. Furthermore, the blurriness intensifies when resizing the element.

The second issue pertains to the use of the InscribedImage constructor. When I attempt to use the line of code var inscribedImage = new joint.shapes.standard.InscribedImage(), I receive a TypeError stating that InscribedImage is not a constructor. This is perplexing as this line of code is directly provided in the JointJS documentation.

I would greatly appreciate any guidance or solutions you could provide to address these issues, particularly any methods that could prevent the images from becoming blurry.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
Could you please provide an example (e.g. JSFiddle) that shows the issue?

- If you want sharp images at all sizes, it is better to provide SVG images (you may have provided a small image that has been resized and has become blurry?).
- Here's an [example](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#shapes.standard) that shows that `InscribedImage` works as expected.

---
*UPDATE:*

There was a typo in the documentation regarding the `InscribedImage` : https://github.com/clientIO/joint/pull/2377

**Reply by VADDIJASWANT** [reply]
![image](https://github.com/clientIO/joint/assets/116190897/8ba77b0a-3e45-49b4-b27d-3b188dab11b0)
![image](https://github.com/clientIO/joint/assets/116190897/efb2880a-8b9a-4ad8-8170-d04773c329b6)

The second image was rendered in my previous product, while the first one was rendered using JointJS. As you can observe, there is a noticeable difference between the two.

In my previous product, the render size and aspect ratio of the image were dynamically adjusted according to the dimensions of the image, resulting in a clear and sharp display. However, in JointJS, the render size and aspect ratio remain constant for all image sizes, causing the image to appear blurry.

Is there a way to dynamically adjust the render size and aspect ratio in JointJS based on the dimensions of the image, similar to my previous product? This adjustment would greatly enhance the clarity of the images rendered in JointJS.

**Reply by kumilingus** [reply]
> the render size and aspect ratio of the image were dynamically adjusted according to the dimensions of the image

I am not sure what do you mean by that

In *JointJS*, the render size and aspect ratio of the exported image do not remain constant. By default, the size of the exported image is equal to the size of the graph content. Note this can be changed by providing a custom `area` to be exported.

There are also the `quality` and `size` options to be used with the `format.Raster` to improve the quality of the `JPEG` image.

Please refer to the plugin documentation: https://resources.jointjs.com/docs/rappid/v3.7/format.Raster.html

But ultimately, I suggest exporting the diagram as SVG, if you don't want to lose the sharpness at any size/zoom level.

**Reply by VADDIJASWANT** [reply]
Hi roman,
 I think there is a small misunderstanding from my communication, I am not trying to export the paper to SVG or JPEG formats, in our product we have a functionality where users can upload their custom images, so we store the image in database in the respective format, and to render that image on the paper we will get the imageURl which can be given as SRC to img tag in html to render that image
From the second image in the above message, you can see in inspect tools that we were rendering the image on our display area using the img tag in our old product
now in JointJS to render that image on the paper I am creating a Standard.Image() graph element and setting the xlink:href property to the imageURL which I am getting from database
you can check the above two images, when I added same image to the new product which I am developing with jointJS and to our own product, the image which is added to the JointJS paper is blurry and is getting blurrier with resizing

**Reply by kumilingus** [reply, marked as answer]
We do not tamper with the image in any way. The low-resolution image must have been provided by you (or your user).

We render [SVG Image](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/image) `<image/>` and you provide value of the `href` attribute.

**Reply by VADDIJASWANT** [reply]
ohh ok Roman, I will look into it from our side, thanks a lot for your help
